.github/workflows/release.yml

```diff
@@ -0,0 +1,46 @@
+name: release
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  build:
+    strategy:
+      fail-fast: true
+      matrix:
+        node:
+          - 18
+        platform:
+          - ubuntu-latest
+    name: "${{matrix.platform}} / Node.js ${{ matrix.node }}"
+    runs-on: ${{matrix.platform}}
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - uses: pnpm/action-setup@v2.4.0
+        with:
+          version: 8
+
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node-version }}
+          registry-url: "https://registry.npmjs.org"
+          cache: "pnpm"
+
+      - run: npm whoami
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+      - run: pnpm install --frozen-lockfile
+
+      - run: pnpm test
+
+      - name: run publish
+        run: pnpm run publish:release
```

Comment:
Any reason we don't just include the lerna command here verbatim? Unless anyone will be calling those targets from outside the workflow, seems like we're spreading across two files unnecessarily.

Reply:
I had a couple reasons for this.

1. I seem to remember that its easier to get the pnpm-installed binaries on your path if you execute a pnpm script first
2. If we ever left github actions for another CI, we just need to change out the actions, not the script itself.
3. I saw this in your PR but the command that wound up in the github workflow used `pnpx` and the command itself was different (and maybe incorrect with how it specified the --canary target?)

Thoughts? I'm not super tied to the way that it is currently built other than point 1 above.

Reply:
Some context on my thinking in each case. Don't feel strongly enough either way to hold up the process though.

1. found some a [few examples](https://stackoverflow.com/questions/57597367/how-to-set-up-github-actions-to-publish-a-lerna-monorepo) where lerna was being called directly from the workflow. Using pnpx was overkill though
2. too much of a "what if" to justify it on it's own
3. relocated that as is from package.json as I didn't think we'd made a decision about canary builds, so agree should have been "minor" instead. I did change the order because (to me) putting "minor" or "major" first reads more clearly as to what it's going to do

Reply:
oh 🤦 `--canary` is just a boolean I thought you had to pass it the `bump` value. If we can hit `lerna` from the CI scripts than I think I agree that its simpler to pull it out of the package.json scripts. Point 2 isn't entirely theoretical, we had to switch from travisci to github actions a while ago. That being said, we're not doing anything too crazy in github actions that should take *that* long to migrate away.

https://www.travis-ci.com/blog/2021-05-07-orgshutdown/